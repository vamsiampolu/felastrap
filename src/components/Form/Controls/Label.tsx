import { PropsWithChildren } from "react";
import { useFela } from "react-fela";

const labelRule = () => ({
  display: "inline-block",
  marginBottom: "0.5rem",
});

export type LabelProps = PropsWithChildren<{
  htmlFor: string;
  className?: string;
}>;

export const Label = ({ htmlFor, children, className = "" }: LabelProps) => {
  const { css } = useFela();
  return (
    <label
      data-testid="form-label"
      htmlFor={htmlFor}
      className={`${css(labelRule)} form-label ${className}`.trim()}
    >
      {children}
    </label>
  );
};
