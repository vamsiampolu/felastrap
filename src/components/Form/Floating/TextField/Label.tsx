import type { ComponentPropsWithoutRef } from "react";
import { useFela } from "react-fela";
import { useFloatingTextContext } from "./useFloatingTextContext";
import { StyleProps } from "../../../common";

export type RuleProps = StyleProps<{
  disabled: boolean;
  placeholder: boolean;
  focused: boolean;
  plainText: boolean;
  readOnly: boolean;
  autoFill: boolean;
}>;

const labelRule = () => ({
  display: "inline-block",
  marginBottom: "0.5rem",
  border: "1px solid transparent",
  height: "100%",
  left: 0,
  overflow: "hidden",
  padding: "1rem 0.75rem",
  pointerEvents: "none",
  position: "absolute",
  textAlign: "start",
  textOverflow: "ellipsis",
  top: 0,
  transformOrigin: "0 0",
  transition: "opacity 0.1s ease-in-out, transform 0.1s ease-in-out",
  whiteSpace: "nowrap",
  zIndex: 2,
  "@media (prefers-reduced-motion: reduce)": {
    transition: "none",
  },
});

const disabledRule = ({ disabled }: RuleProps) =>
  disabled
    ? {
        color: "#6c757d",
        "::after": {
          backgroundColor: "#e9ecef",
        },
      }
    : {};

const noPlaceholderRule = ({ placeholder }: RuleProps) =>
  placeholder
    ? {
        transform: "scale(0.85) translateY(0.5rem) translateX(0.15rem)",
      }
    : {
        color: "rgba(33, 37, 41, 0.65)",
        transform: "scale(0.85) translateY(-0.5rem) translateX(0.15rem)",
        "::after": {
          position: "absolute",
          inset: "1rem 0.375rem",
          zIndex: -1,
          height: "1.5em",
          content: "",
          backgroundColor: "#fff",
          borderRadius: "0.375rem",
        },
      };

const focusRule = ({ focused }: RuleProps) =>
  focused
    ? {
        color: "rgba(33, 37, 41, 0.65)",
        transform:
          "scale(0.85) translateY(-0.5rem) translateX(0.15rem) !important",
        "::after": {
          position: "absolute",
          inset: "1rem 0.375rem",
          zIndex: -1,
          height: "1.5em",
          content: "",
          backgroundColor: "#fff",
          borderRadius: "0.375rem",
        },
      }
    : {};

const plainTextReadOnlyRule = ({ plainText, readOnly }: RuleProps) =>
  plainText && readOnly
    ? {
        borderWidth: "1px 0",
        color: "rgba(33, 37, 41, 0.65)",
        transform: "scale(0.85) translateY(-0.5rem) translateX(0.15rem)",
        "::after": {
          position: "absolute",
          inset: "1rem 0.375rem",
          zIndex: -1,
          height: "1.5em",
          content: "",
          backgroundColor: "#fff",
          borderRadius: "0.375rem",
        },
      }
    : {};

const autofillRule = ({ autoFill }: RuleProps) => {
  if (autoFill) {
    return {
      color: "rgba(33, 37, 41, 0.65)",
      transform: "scale(0.85) translateY(-0.5rem) translateX(0.15rem)",
    };
  }

  return {};
};

export type FloatingLabelProps = Omit<
  ComponentPropsWithoutRef<"label">,
  "children"
> & { label?: string };

export const FloatingLabel = ({
  className = "",
  label,
  htmlFor,
  ...props
}: FloatingLabelProps) => {
  const {
    autoFilled: autoFill = false,
    disabled = false,
    focused = false,
    placeholderShown: placeholder = false,
    plainText = false,
    readOnly = false,
  } = useFloatingTextContext();

  const { css } = useFela({
    autoFill,
    disabled,
    focused,
    placeholder,
    plainText,
    readOnly,
  });

  const labelClassName = `${css([
    labelRule,
    focusRule,
    disabledRule,
    noPlaceholderRule,
    plainTextReadOnlyRule,
    autofillRule,
  ])} floating-label ${className}`.trim();

  return (
    <label
      data-testid="floating-label"
      htmlFor={htmlFor}
      className={labelClassName}
      {...props}
    >
      {label}
    </label>
  );
};
