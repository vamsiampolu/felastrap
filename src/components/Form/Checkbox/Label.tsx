import { useFela } from "react-fela";
import { LabelProps, Label } from "../Controls/Label";
import { StyleProps } from "../../common";

export type CheckboxLabelProps = LabelProps & { disabled?: boolean };

const disabledRule = ({ disabled }: StyleProps<{ disabled: boolean }>) =>
  disabled
    ? {
        cursor: "default",
        opacity: 0.5,
      }
    : {};

export const CheckboxLabel = ({
  disabled = false,
  className = "",
  htmlFor,
  children,
}: CheckboxLabelProps) => {
  const { css } = useFela({ disabled });
  const cx = `${css(disabledRule)} form-check-label ${className}`.trim();
  return (
    <Label htmlFor={htmlFor} className={cx}>
      {children}
    </Label>
  );
};
