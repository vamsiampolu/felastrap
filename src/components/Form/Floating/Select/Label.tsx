import { useFela } from "react-fela";
import { Label, LabelProps } from "../../Controls/Label";
import { StyleProps } from "../../../common";

const rule = () => ({
  position: "absolute",
  top: 0,
  left: 0,
  zIndex: 2,
  height: "100%",
  padding: "1rem 0.75rem",
  overflow: "hidden",
  textAlign: "start",
  textOverflow: "ellipsis",
  whiteSpace: "nowrap",
  pointerEvents: "none",
  border: "1px solid transparent",
  transformOrigin: "0 0",
  transition: "opacity 0.1s ease-in-out, transform 0.1s ease-in-out",
  color: "rgba(33, 37, 41, 0.65)",
  transform: "scale(0.85) translateY(-0.5rem) translateX(0.15rem)",
  "@media (prefers-reduced-motion: reduce)": {
    transition: "none",
  },
  "::after": {
    position: "absolute",
    inset: "1rem 0.375rem",
    zIndex: -1,
    height: "1.5em",
    content: "",
    backgroundColor: "#fff",
    borderRadius: "0.375rem",
  },
});

const disabledRule = ({ disabled }: StyleProps<{ disabled: boolean }>) =>
  disabled
    ? {
        color: "#6c757d",
      }
    : {};

export type FloatingSelectLabelProps = LabelProps & { disabled?: boolean };

export const FloatingSelectLabel = ({
  disabled = false,
  children,
  className = "",
  ...props
}: FloatingSelectLabelProps) => {
  const { css } = useFela({ disabled });
  const cx = `${css([
    rule,
    disabledRule,
  ])} floating-select-label ${className}`.trim();
  return (
    <Label className={cx} {...props}>
      {children}
    </Label>
  );
};
