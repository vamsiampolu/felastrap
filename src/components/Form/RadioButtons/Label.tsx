import { useFela } from "react-fela";
import { Label, type LabelProps } from "../Controls/Label";
import { StyleProps } from "../../common";

const disabledRule = ({ disabled }: StyleProps<{ disabled: boolean }>) =>
  disabled ? { cursor: "default", opacity: "0.5" } : {};

export type RadioLabelProps = LabelProps & { disabled?: boolean };

export const RadioLabel = ({
  className = "",
  children,
  htmlFor,
  disabled = false,
}: RadioLabelProps) => {
  const { css } = useFela({ disabled });
  const cx = `${css(disabledRule)} form-check-label ${className}`.trim();
  return (
    <Label className={cx} htmlFor={htmlFor}>
      {children}
    </Label>
  );
};
