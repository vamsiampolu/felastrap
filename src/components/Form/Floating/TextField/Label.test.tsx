import { render, screen } from "@testing-library/react";
import { IRenderer } from "fela";
import {
  Wrapper,
  createRenderer,
  createSnapshot,
  cssReboot,
} from "../../../../test-helpers";
import {
  FloatingTextContext,
  FloatingTextContextValue,
} from "./FloatingTextContext";
import { FloatingLabel } from "./Label";

describe("components/Form/Floating/TextField/Label", () => {
  let renderer: IRenderer;

  beforeEach(() => {
    renderer = createRenderer();
    cssReboot(renderer);
  });

  afterEach(() => {
    renderer.clear();
  });

  it("should display a floating input label", async () => {
    const component = <FloatingLabel label="Floating Label" />;
    render(<Wrapper renderer={renderer}>{component}</Wrapper>);

    expect(screen.getByTestId("floating-label")).toBeInTheDocument();
    expect(screen.queryByTestId("floating-label")).toHaveTextContent(
      "Floating Label",
    );
    expect(screen.queryByTestId("floating-label")).toHaveClass(
      "floating-label",
    );

    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });

  it("should display a disabled floating input label", async () => {
    const value = {
      autoFilled: false,
      disabled: true,
      focused: false,
      placeholderShown: false,
      plainText: false,
      readOnly: false,
    } as FloatingTextContextValue;

    const component = (
      <FloatingTextContext.Provider value={value}>
        <FloatingLabel htmlFor="foobar" label="Floating Label" />
      </FloatingTextContext.Provider>
    );

    render(<Wrapper renderer={renderer}>{component}</Wrapper>);

    expect(screen.getByTestId("floating-label")).toBeInTheDocument();
    expect(screen.queryByTestId("floating-label")).toHaveAttribute(
      "for",
      "foobar",
    );

    expect(screen.queryByTestId("floating-label")).toHaveClass(
      "floating-label",
    );
    expect(screen.queryByTestId("floating-label")).toHaveTextContent(
      "Floating Label",
    );

    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });

  it("should display a plaintext readOnly floating input label", async () => {
    const value = {
      autoFilled: false,
      disabled: false,
      focused: false,
      placeholderShown: false,
      plainText: true,
      readOnly: true,
    } as FloatingTextContextValue;

    const component = (
      <FloatingTextContext.Provider value={value}>
        <FloatingLabel htmlFor="foobar" label="Floating Label" />
      </FloatingTextContext.Provider>
    );

    render(<Wrapper renderer={renderer}>{component}</Wrapper>);

    expect(screen.getByTestId("floating-label")).toBeInTheDocument();
    expect(screen.queryByTestId("floating-label")).toHaveAttribute(
      "for",
      "foobar",
    );

    expect(screen.queryByTestId("floating-label")).toHaveClass(
      "floating-label",
    );
    expect(screen.queryByTestId("floating-label")).toHaveTextContent(
      "Floating Label",
    );

    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });

  it("should display the label for autofilled floating input", async () => {
    const value = {
      autoFilled: true,
      disabled: false,
      focused: false,
      placeholderShown: false,
      plainText: false,
      readOnly: false,
    } as FloatingTextContextValue;

    const component = (
      <FloatingTextContext.Provider value={value}>
        <FloatingLabel htmlFor="foobar" label="Floating Label" />
      </FloatingTextContext.Provider>
    );

    render(<Wrapper renderer={renderer}>{component}</Wrapper>);

    expect(screen.getByTestId("floating-label")).toBeInTheDocument();
    expect(screen.queryByTestId("floating-label")).toHaveAttribute(
      "for",
      "foobar",
    );

    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });

  it("should show the label for a focused floating input", async () => {
    const value = {
      autoFilled: false,
      disabled: false,
      focused: true,
      placeholderShown: false,
      plainText: false,
      readOnly: false,
    } as FloatingTextContextValue;

    const component = (
      <FloatingTextContext.Provider value={value}>
        <FloatingLabel htmlFor="foobar" label="Floating Label" />
      </FloatingTextContext.Provider>
    );

    render(<Wrapper renderer={renderer}>{component}</Wrapper>);

    expect(screen.getByTestId("floating-label")).toBeInTheDocument();
    expect(screen.queryByTestId("floating-label")).toHaveAttribute(
      "for",
      "foobar",
    );

    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });

  it("should show a label when placeholder is visible for a floating input", async () => {
    const value = {
      autoFilled: false,
      disabled: false,
      focused: false,
      placeholderShown: true,
      plainText: false,
      readOnly: false,
    } as FloatingTextContextValue;

    const component = (
      <FloatingTextContext.Provider value={value}>
        <FloatingLabel htmlFor="foobar" label="Floating Label" />
      </FloatingTextContext.Provider>
    );

    render(<Wrapper renderer={renderer}>{component}</Wrapper>);

    expect(screen.getByTestId("floating-label")).toBeInTheDocument();
    expect(screen.queryByTestId("floating-label")).toHaveAttribute(
      "for",
      "foobar",
    );

    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });
});
