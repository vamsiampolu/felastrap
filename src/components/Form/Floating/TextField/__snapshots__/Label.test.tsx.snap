// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Form/Floating/TextField/Label should display a disabled floating input label 1`] = `
".a {
  display: inline-block;
}
.b {
  margin-bottom: 0.5rem;
}
.c {
  height: 100%;
}
.d {
  left: 0;
}
.e {
  pointer-events: none;
}
.f {
  position: absolute;
}
.g {
  text-align: start;
}
.h {
  text-overflow: ellipsis;
}
.i {
  top: 0;
}
.j {
  transform-origin: 0 0;
}
.k {
  white-space: nowrap;
}
.l {
  z-index: 2;
}
.q {
  color: rgba(33, 37, 41, 0.65);
}
.r::after {
  background-color: #fff;
}
.s::after {
  position: absolute;
}
.t::after {
  z-index: -1;
}
.u::after {
  height: 1.5em;
}
.v::after {
  content:;
}
.w::after {
  border-top-left-radius: 0.375rem;
}
.x::after {
  border-top-right-radius: 0.375rem;
}
.y::after {
  border-bottom-right-radius: 0.375rem;
}
.z::after {
  border-bottom-left-radius: 0.375rem;
}
.ab::after {
  top: 1rem;
}
.ac::after {
  right: 0.375rem;
}
.ae::after {
  bottom: 1rem;
}
.af::after {
  left: 0.375rem;
}
.ag {
  transform: scale(0.85) translateY(-0.5rem) translateX(0.15rem);
}
.ah {
  transition-property: opacity, transform;
}
.ai {
  transition-duration: 0.1s, 0.1s;
}
.aj {
  transition-delay: ease-in-out, ease-in-out;
}
.ak {
  transition-timing-function: ease, ease;
}
.al {
  padding-top: 1rem;
}
.am {
  padding-right: 0.75rem;
}
.an {
  padding-bottom: 1rem;
}
.ao {
  padding-left: 0.75rem;
}
.ap {
  overflow-x: hidden;
}
.aq {
  overflow-y: hidden;
}
.ar {
  border-top-width: 1px;
}
.as {
  border-right-width: 1px;
}
.at {
  border-bottom-width: 1px;
}
.au {
  border-left-width: 1px;
}
.av {
  border-top-style: solid;
}
.aw {
  border-right-style: solid;
}
.ax {
  border-bottom-style: solid;
}
.ay {
  border-left-style: solid;
}
.az {
  border-top-color: transparent;
}
.ba {
  border-right-color: transparent;
}
.bb {
  border-bottom-color: transparent;
}
.bc {
  border-left-color: transparent;
}
@media (prefers-reduced-motion: reduce) {
  .m {
    transition-property: none;
  }
  .n {
    transition-duration: 0s;
  }
  .o {
    transition-delay: 0s;
  }
  .p {
    transition-timing-function: ease;
  }
}


<label
  data-testid="floating-label"
  for="foobar"
  class="a ab ac ae af ag ah ai aj ak al am an ao ap aq ar as at au av aw ax ay az b ba bb bc c d e f g h i j k l m n o p q r s t u v w x y z floating-label"
  >Floating Label</label
>;
"
`;

exports[`components/Form/Floating/TextField/Label should display a floating input label 1`] = `
".a {
  display: inline-block;
}
.b {
  margin-bottom: 0.5rem;
}
.c {
  height: 100%;
}
.d {
  left: 0;
}
.e {
  pointer-events: none;
}
.f {
  position: absolute;
}
.g {
  text-align: start;
}
.h {
  text-overflow: ellipsis;
}
.i {
  top: 0;
}
.j {
  transform-origin: 0 0;
}
.k {
  white-space: nowrap;
}
.l {
  z-index: 2;
}
.q {
  color: rgba(33, 37, 41, 0.65);
}
.r {
  transform: scale(0.85) translateY(-0.5rem) translateX(0.15rem);
}
.s::after {
  position: absolute;
}
.t::after {
  z-index: -1;
}
.u::after {
  height: 1.5em;
}
.v::after {
  content:;
}
.w::after {
  background-color: #fff;
}
.x::after {
  border-top-left-radius: 0.375rem;
}
.y::after {
  border-top-right-radius: 0.375rem;
}
.z::after {
  border-bottom-right-radius: 0.375rem;
}
.ab::after {
  border-bottom-left-radius: 0.375rem;
}
.ac::after {
  top: 1rem;
}
.ae::after {
  right: 0.375rem;
}
.af::after {
  bottom: 1rem;
}
.ag::after {
  left: 0.375rem;
}
.ah {
  transition-property: opacity, transform;
}
.ai {
  transition-duration: 0.1s, 0.1s;
}
.aj {
  transition-delay: ease-in-out, ease-in-out;
}
.ak {
  transition-timing-function: ease, ease;
}
.al {
  padding-top: 1rem;
}
.am {
  padding-right: 0.75rem;
}
.an {
  padding-bottom: 1rem;
}
.ao {
  padding-left: 0.75rem;
}
.ap {
  overflow-x: hidden;
}
.aq {
  overflow-y: hidden;
}
.ar {
  border-top-width: 1px;
}
.as {
  border-right-width: 1px;
}
.at {
  border-bottom-width: 1px;
}
.au {
  border-left-width: 1px;
}
.av {
  border-top-style: solid;
}
.aw {
  border-right-style: solid;
}
.ax {
  border-bottom-style: solid;
}
.ay {
  border-left-style: solid;
}
.az {
  border-top-color: transparent;
}
.ba {
  border-right-color: transparent;
}
.bb {
  border-bottom-color: transparent;
}
.bc {
  border-left-color: transparent;
}
@media (prefers-reduced-motion: reduce) {
  .m {
    transition-property: none;
  }
  .n {
    transition-duration: 0s;
  }
  .o {
    transition-delay: 0s;
  }
  .p {
    transition-timing-function: ease;
  }
}


<label
  data-testid="floating-label"
  class="a ab ac ae af ag ah ai aj ak al am an ao ap aq ar as at au av aw ax ay az b ba bb bc c d e f g h i j k l m n o p q r s t u v w x y z floating-label"
  >Floating Label</label
>;
"
`;

exports[`components/Form/Floating/TextField/Label should display a plaintext readOnly floating input label 1`] = `
".a {
  display: inline-block;
}
.b {
  margin-bottom: 0.5rem;
}
.c {
  height: 100%;
}
.d {
  left: 0;
}
.e {
  pointer-events: none;
}
.f {
  position: absolute;
}
.g {
  text-align: start;
}
.h {
  text-overflow: ellipsis;
}
.i {
  top: 0;
}
.j {
  transform-origin: 0 0;
}
.k {
  white-space: nowrap;
}
.l {
  z-index: 2;
}
.q {
  color: rgba(33, 37, 41, 0.65);
}
.r {
  transform: scale(0.85) translateY(-0.5rem) translateX(0.15rem);
}
.s::after {
  position: absolute;
}
.t::after {
  z-index: -1;
}
.u::after {
  height: 1.5em;
}
.v::after {
  content:;
}
.w::after {
  background-color: #fff;
}
.x::after {
  border-top-left-radius: 0.375rem;
}
.y::after {
  border-top-right-radius: 0.375rem;
}
.z::after {
  border-bottom-right-radius: 0.375rem;
}
.ab::after {
  border-bottom-left-radius: 0.375rem;
}
.ac::after {
  top: 1rem;
}
.ae::after {
  right: 0.375rem;
}
.af::after {
  bottom: 1rem;
}
.ag::after {
  left: 0.375rem;
}
.ah {
  border-top-width: 1px;
}
.ai {
  border-right-width: 0;
}
.aj {
  border-bottom-width: 1px;
}
.ak {
  border-left-width: 0;
}
.al {
  transition-property: opacity, transform;
}
.am {
  transition-duration: 0.1s, 0.1s;
}
.an {
  transition-delay: ease-in-out, ease-in-out;
}
.ao {
  transition-timing-function: ease, ease;
}
.ap {
  padding-top: 1rem;
}
.aq {
  padding-right: 0.75rem;
}
.ar {
  padding-bottom: 1rem;
}
.as {
  padding-left: 0.75rem;
}
.at {
  overflow-x: hidden;
}
.au {
  overflow-y: hidden;
}
.av {
  border-top-style: solid;
}
.aw {
  border-right-style: solid;
}
.ax {
  border-bottom-style: solid;
}
.ay {
  border-left-style: solid;
}
.az {
  border-top-color: transparent;
}
.ba {
  border-right-color: transparent;
}
.bb {
  border-bottom-color: transparent;
}
.bc {
  border-left-color: transparent;
}
@media (prefers-reduced-motion: reduce) {
  .m {
    transition-property: none;
  }
  .n {
    transition-duration: 0s;
  }
  .o {
    transition-delay: 0s;
  }
  .p {
    transition-timing-function: ease;
  }
}


<label
  data-testid="floating-label"
  for="foobar"
  class="a ab ac ae af ag ah ai aj ak al am an ao ap aq ar as at au av aw ax ay az b ba bb bc c d e f g h i j k l m n o p q r s t u v w x y z floating-label"
  >Floating Label</label
>;
"
`;

exports[`components/Form/Floating/TextField/Label should display the label for autofilled floating input 1`] = `
".a {
  display: inline-block;
}
.b {
  margin-bottom: 0.5rem;
}
.c {
  height: 100%;
}
.d {
  left: 0;
}
.e {
  pointer-events: none;
}
.f {
  position: absolute;
}
.g {
  text-align: start;
}
.h {
  text-overflow: ellipsis;
}
.i {
  top: 0;
}
.j {
  transform-origin: 0 0;
}
.k {
  white-space: nowrap;
}
.l {
  z-index: 2;
}
.q {
  color: rgba(33, 37, 41, 0.65);
}
.r {
  transform: scale(0.85) translateY(-0.5rem) translateX(0.15rem);
}
.s::after {
  position: absolute;
}
.t::after {
  z-index: -1;
}
.u::after {
  height: 1.5em;
}
.v::after {
  content:;
}
.w::after {
  background-color: #fff;
}
.x::after {
  border-top-left-radius: 0.375rem;
}
.y::after {
  border-top-right-radius: 0.375rem;
}
.z::after {
  border-bottom-right-radius: 0.375rem;
}
.ab::after {
  border-bottom-left-radius: 0.375rem;
}
.ac::after {
  top: 1rem;
}
.ae::after {
  right: 0.375rem;
}
.af::after {
  bottom: 1rem;
}
.ag::after {
  left: 0.375rem;
}
.ah {
  transition-property: opacity, transform;
}
.ai {
  transition-duration: 0.1s, 0.1s;
}
.aj {
  transition-delay: ease-in-out, ease-in-out;
}
.ak {
  transition-timing-function: ease, ease;
}
.al {
  padding-top: 1rem;
}
.am {
  padding-right: 0.75rem;
}
.an {
  padding-bottom: 1rem;
}
.ao {
  padding-left: 0.75rem;
}
.ap {
  overflow-x: hidden;
}
.aq {
  overflow-y: hidden;
}
.ar {
  border-top-width: 1px;
}
.as {
  border-right-width: 1px;
}
.at {
  border-bottom-width: 1px;
}
.au {
  border-left-width: 1px;
}
.av {
  border-top-style: solid;
}
.aw {
  border-right-style: solid;
}
.ax {
  border-bottom-style: solid;
}
.ay {
  border-left-style: solid;
}
.az {
  border-top-color: transparent;
}
.ba {
  border-right-color: transparent;
}
.bb {
  border-bottom-color: transparent;
}
.bc {
  border-left-color: transparent;
}
@media (prefers-reduced-motion: reduce) {
  .m {
    transition-property: none;
  }
  .n {
    transition-duration: 0s;
  }
  .o {
    transition-delay: 0s;
  }
  .p {
    transition-timing-function: ease;
  }
}


<label
  data-testid="floating-label"
  for="foobar"
  class="a ab ac ae af ag ah ai aj ak al am an ao ap aq ar as at au av aw ax ay az b ba bb bc c d e f g h i j k l m n o p q r s t u v w x y z floating-label"
  >Floating Label</label
>;
"
`;

exports[`components/Form/Floating/TextField/Label should show a label when placeholder is visible for a floating input 1`] = `
".a {
  display: inline-block;
}
.b {
  margin-bottom: 0.5rem;
}
.c {
  height: 100%;
}
.d {
  left: 0;
}
.e {
  pointer-events: none;
}
.f {
  position: absolute;
}
.g {
  text-align: start;
}
.h {
  text-overflow: ellipsis;
}
.i {
  top: 0;
}
.j {
  transform-origin: 0 0;
}
.k {
  white-space: nowrap;
}
.l {
  z-index: 2;
}
.q {
  transform: scale(0.85) translateY(0.5rem) translateX(0.15rem);
}
.r {
  transition-property: opacity, transform;
}
.s {
  transition-duration: 0.1s, 0.1s;
}
.t {
  transition-delay: ease-in-out, ease-in-out;
}
.u {
  transition-timing-function: ease, ease;
}
.v {
  padding-top: 1rem;
}
.w {
  padding-right: 0.75rem;
}
.x {
  padding-bottom: 1rem;
}
.y {
  padding-left: 0.75rem;
}
.z {
  overflow-x: hidden;
}
.ab {
  overflow-y: hidden;
}
.ac {
  border-top-width: 1px;
}
.ae {
  border-right-width: 1px;
}
.af {
  border-bottom-width: 1px;
}
.ag {
  border-left-width: 1px;
}
.ah {
  border-top-style: solid;
}
.ai {
  border-right-style: solid;
}
.aj {
  border-bottom-style: solid;
}
.ak {
  border-left-style: solid;
}
.al {
  border-top-color: transparent;
}
.am {
  border-right-color: transparent;
}
.an {
  border-bottom-color: transparent;
}
.ao {
  border-left-color: transparent;
}
@media (prefers-reduced-motion: reduce) {
  .m {
    transition-property: none;
  }
  .n {
    transition-duration: 0s;
  }
  .o {
    transition-delay: 0s;
  }
  .p {
    transition-timing-function: ease;
  }
}


<label
  data-testid="floating-label"
  for="foobar"
  class="a ab ac ae af ag ah ai aj ak al am an ao b c d e f g h i j k l m n o p q r s t u v w x y z floating-label"
  >Floating Label</label
>;
"
`;

exports[`components/Form/Floating/TextField/Label should show the label for a focused floating input 1`] = `
".a {
  display: inline-block;
}
.b {
  margin-bottom: 0.5rem;
}
.c {
  height: 100%;
}
.d {
  left: 0;
}
.e {
  pointer-events: none;
}
.f {
  position: absolute;
}
.g {
  text-align: start;
}
.h {
  text-overflow: ellipsis;
}
.i {
  top: 0;
}
.j {
  transform-origin: 0 0;
}
.k {
  white-space: nowrap;
}
.l {
  z-index: 2;
}
.q {
  color: rgba(33, 37, 41, 0.65);
}
.r {
  transform: scale(0.85) translateY(-0.5rem) translateX(0.15rem);
}
.s::after {
  position: absolute;
}
.t::after {
  z-index: -1;
}
.u::after {
  height: 1.5em;
}
.v::after {
  content:;
}
.w::after {
  background-color: #fff;
}
.x::after {
  border-top-left-radius: 0.375rem;
}
.y::after {
  border-top-right-radius: 0.375rem;
}
.z::after {
  border-bottom-right-radius: 0.375rem;
}
.ab::after {
  border-bottom-left-radius: 0.375rem;
}
.ac::after {
  top: 1rem;
}
.ae::after {
  right: 0.375rem;
}
.af::after {
  bottom: 1rem;
}
.ag::after {
  left: 0.375rem;
}
.ah {
  transition-property: opacity, transform;
}
.ai {
  transition-duration: 0.1s, 0.1s;
}
.aj {
  transition-delay: ease-in-out, ease-in-out;
}
.ak {
  transition-timing-function: ease, ease;
}
.al {
  padding-top: 1rem;
}
.am {
  padding-right: 0.75rem;
}
.an {
  padding-bottom: 1rem;
}
.ao {
  padding-left: 0.75rem;
}
.ap {
  overflow-x: hidden;
}
.aq {
  overflow-y: hidden;
}
.ar {
  border-top-width: 1px;
}
.as {
  border-right-width: 1px;
}
.at {
  border-bottom-width: 1px;
}
.au {
  border-left-width: 1px;
}
.av {
  border-top-style: solid;
}
.aw {
  border-right-style: solid;
}
.ax {
  border-bottom-style: solid;
}
.ay {
  border-left-style: solid;
}
.az {
  border-top-color: transparent;
}
.ba {
  border-right-color: transparent;
}
.bb {
  border-bottom-color: transparent;
}
.bc {
  border-left-color: transparent;
}
@media (prefers-reduced-motion: reduce) {
  .m {
    transition-property: none;
  }
  .n {
    transition-duration: 0s;
  }
  .o {
    transition-delay: 0s;
  }
  .p {
    transition-timing-function: ease;
  }
}


<label
  data-testid="floating-label"
  for="foobar"
  class="a ab ac ae af ag ah ai aj ak al am an ao ap aq ar as at au av aw ax ay az b ba bb bc c d e f g h i j k l m n o p q r s t u v w x y z floating-label"
  >Floating Label</label
>;
"
`;
